api/revoke.go

```diff
@@ -0,0 +1,137 @@
+package api
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/smallstep/certificates/authority"
+	"golang.org/x/crypto/ocsp"
+)
+
+// RevocationReasonCodes is a map between string reason codes
+// to integers as defined in RFC 5280
+var RevocationReasonCodes = map[string]int{
+	"unspecified":          ocsp.Unspecified,
+	"keycompromise":        ocsp.KeyCompromise,
+	"cacompromise":         ocsp.CACompromise,
+	"affiliationchanged":   ocsp.AffiliationChanged,
+	"superseded":           ocsp.Superseded,
+	"cessationofoperation": ocsp.CessationOfOperation,
+	"certificatehold":      ocsp.CertificateHold,
+	"removefromcrl":        ocsp.RemoveFromCRL,
+	"privilegewithdrawn":   ocsp.PrivilegeWithdrawn,
+	"aacompromise":         ocsp.AACompromise,
+}
+
+// ReasonStringToCode tries to convert a reason string to an integer code
+func ReasonStringToCode(reason string) (int, error) {
+	// default to 0
+	if reason == "" {
+		return 0, nil
+	}
+
+	code, found := RevocationReasonCodes[strings.ToLower(reason)]
```

Comment:
We can also strip spaces so a user can write the reason in a more natural language, i.e: 
`--reason 'affiliation changed'`